tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -110,3 +118,104 @@ def test_write(self):
+                             node_prealloc_count=1,
+                             num_brokers=7,
+                             si_settings=si_settings)
+
+    def setUp(self):
+        # Dedicated nodes refers to non-container nodes such as EC2 instances
+        self.topics[
+            0].partition_count = 1000 if self.redpanda.dedicated_nodes else 100
+
+        # Topic creation happens here
+        super().setUp()
+
+        # Remote write/read and retention set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+
+    @cluster(num_nodes=8)
+    def test_create_or_delete_topics_while_busy(self):
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        # 100k messages of size 2**18
+        # is ~24GB of data. 500k messages
+        # of size 2**18 is ~122GB of data.
+        msg_size = 2**18
+        msg_count = 500000 if self.redpanda.dedicated_nodes else 100000
+        timeout = 600
+
+        producer = FranzGoVerifiableProducer(self.test_context, self.redpanda,
+                                             self.topic, msg_size, msg_count,
+                                             self.preallocated_nodes)
+        producer.start(clean=False)
+        # Block until a subset of records are produced
+        await_minimum_produced_records(self.redpanda,
+                                       producer,
+                                       min_acked=msg_count / 100)
+
+        rand_consumer = FranzGoVerifiableRandomConsumer(
+            self.test_context, self.redpanda, self.topic, msg_size, 100, 10,
+            self.preallocated_nodes)
+
+        rand_consumer.start(clean=False)
+
+        random_topics = []
+
+        # Do random ops until the producer stops
+        def create_or_delete_until_producer_fin():
+            nonlocal random_topics
+            trigger = random.randint(1, 2)
+
+            if trigger == 1:
+                some_topic = TopicSpec()
+                self.logger.debug(f'Create topic: {some_topic}')
+                self.client().create_topic(some_topic)
+                random_topics.append(some_topic)
+
+            if trigger == 2:
+                if len(random_topics) > 0:
+                    random.shuffle(random_topics)
+                    some_topic = random_topics.pop()
+                    self.logger.debug(f'Delete topic: {some_topic}')
+                    self.client().delete_topic(some_topic.name)
+
+            return producer.status == ServiceStatus.FINISH
+
+        wait_until(create_or_delete_until_producer_fin,
```

Comment:
I think that this needs a comment that says that the predicate in `wait_until` is used not only to wait but also to apply some changes.